approvals/src/test/java/no/sikt/nva/approvals/persistence/DynamoDbApprovalRepositoryTest.java

```diff
@@ -149,6 +151,89 @@ void shouldSaveAndFindIdentifiersWhenMoreThan100Provided() throws RepositoryExce
         assertEquals(identifiers.size(), persistedIdentifiers.size());
     }
 
+    @Test
+    void shouldUpdateApprovalIdentifiersByAddingNewIdentifiers() throws RepositoryException {
+        var initialIdentifiers = randomIdentifiers(2);
+        var approval = randomApproval(initialIdentifiers, randomUUID());
+        approvalRepository.save(approval);
+
+        var newIdentifiers = randomIdentifiers(3);
+        var allIdentifiers = new java.util.ArrayList<>(initialIdentifiers);
+        allIdentifiers.addAll(newIdentifiers);
+        var updatedApproval = new Approval(approval.identifier(), allIdentifiers, approval.source(), approval.handle());
+
+        approvalRepository.updateApprovalIdentifiers(updatedApproval);
+        var persistedApproval = approvalRepository.findByApprovalIdentifier(approval.identifier());
+
+        assertTrue(persistedApproval.orElseThrow().namedIdentifiers().containsAll(allIdentifiers));
+    }
+
+    @Test
+    void shouldUpdateApprovalIdentifiersByRemovingIdentifiers() throws RepositoryException {
+        var initialIdentifiers = randomIdentifiers(5);
+        var approval = randomApproval(initialIdentifiers, randomUUID());
+        approvalRepository.save(approval);
+
+        var remainingIdentifiers = initialIdentifiers.stream().limit(2).toList();
+        var updatedApproval = new Approval(approval.identifier(), remainingIdentifiers, approval.source(),
+            approval.handle());
+
+        approvalRepository.updateApprovalIdentifiers(updatedApproval);
+        var persistedApproval = approvalRepository.findByApprovalIdentifier(approval.identifier());
+
+        assertTrue(persistedApproval.orElseThrow().namedIdentifiers().containsAll(remainingIdentifiers));
+    }
+
+    @Test
+    void shouldUpdateApprovalIdentifiersByAddingAndRemovingIdentifiers() throws RepositoryException {
+        var initialIdentifiers = randomIdentifiers(3);
+        var approval = randomApproval(initialIdentifiers, randomUUID());
+        approvalRepository.save(approval);
+
+        var keptIdentifiers = initialIdentifiers.stream().limit(1).toList();
+        var newIdentifiers = randomIdentifiers(2);
+        var finalIdentifiers = new ArrayList<>(keptIdentifiers);
+        finalIdentifiers.addAll(newIdentifiers);
+
+        var updatedApproval = new Approval(approval.identifier(), finalIdentifiers, approval.source(), approval.handle());
+
+        approvalRepository.updateApprovalIdentifiers(updatedApproval);
+        var persistedApproval = approvalRepository.findByApprovalIdentifier(approval.identifier());
+
+        assertTrue(persistedApproval.orElseThrow().namedIdentifiers().containsAll(finalIdentifiers));
+    }
+
+    @Test
+    void shouldUpdateApprovalIdentifiersWhenApprovalHasMoreThan80Identifiers() throws RepositoryException {
```

Comment:
I have 50 identifiers, remove 40, add 60. That is 70 that we keep, probably 110 transactions/writes. not sure about the title of this test